staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -796,24 +804,67 @@ type ruleCost struct {
 	Cost uint64
 }
 
+// MultiplyByElementCost returns a costInfo where the MaxCardinality is multiplied by the
+// factor that the schema increases the cardinality of its children. If the costInfo's
+// MaxCardinality is unbounded (nil) or the factor that the schema increase the cardinality
+// is unbounded, the resulting costInfo's MaxCardinality is also unbounded.
 func (c *costInfo) MultiplyByElementCost(schema *apiextensions.JSONSchemaProps) costInfo {
 	result := costInfo{CRDCost: c.CRDCost}
```

Comment:
this might make it clearer we're starting with an unbounded return value


```suggestion
	result := costInfo{CRDCost: c.CRDCost, MaxCardinality: unbounded}
```

Reply:
Thanks, yes. I missed that.